Python/thread_pthread.h

```diff
@@ -152,6 +152,24 @@ PyThread__init_thread(void)
  * Thread support.
  */
 
+typedef struct {
+    void (*func) (void *);
+    void *arg;
+} pthread_func;
+
+static void *
+pthread_helper_func(void *fn)
+{
+    pthread_func *pfn = fn;
+
+    void (*func) (void *) = pfn->func;
+    void *arg = pfn->arg;
+
+    func(arg);
+
+    free (fn);
```

Comment:
Please use `PyMem_RawMalloc/PyMem_RawFree` like in other parts of the module (I've used malloc as a generic term before, sorry).

It's better to free it before calling `func`.

Please follow [PEP 7](https://www.python.org/dev/peps/pep-0007) here and in other places (here, there should be no spaces between the function name and the parenthesis).